ResearchKit/Common/ORKPasscodeStepViewController.m

```diff
@@ -369,7 +369,17 @@ - (void)promptTouchId {
                                                             }]];
                     [strongSelf presentViewController:alert animated:YES completion:nil];
                 } else if (error.code == LAErrorUserCancel) {
-                    [strongSelf makePasscodeViewBecomeFirstResponder];
+
+                    // call becomeFirstResponder here to show the keyboard. dispatch to main queue with
```

Comment:
Yes, I did dispatch without delay and it worked. I added a delay only because I thought it might be safer and and it seemed to make for a smoother transition. Based on what you read, it seems it makes no difference from a reliability perspective.